### PR TITLE
session: add check resource group hint existence

### LINKED ISSUE #45178: Query execution failed with not exist ResourceGroup() hint
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
use test;
create table t (id int);
select /*+ RESOURCE_GROUP(test) */ * from t;
```

### 2. What did you expect to see? (Required)
Query should success because hint should not affect the result's correctness.

### 3. What did you see instead (Required)
```
ERROR 8249 (HY000): Unknown resource group 'test'
```

### 4. What is your TiDB version? (Required)
nightly

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close #45178

Problem Summary:

### What is changed and how it works?
After setting resource group name from hint, check the group name's existence and directly fallback to the original group name if the hint name is invalid.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]
lgtm

File: session/session.go [hunk 1]
```diff
@@ -2226,11 +2226,22 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
...
+				kv.SetTxnResourceGroup(txn, originalResourceGroup)
+			}
+		} else {
+			defer func() {
+				// Restore the resource group for the session
+				sessVars.ResourceGroupName = originalResourceGroup
+			}()
+		}
 	}
+	s.GetDomainInfoSchema()
```
Comment (CabinfeverB) [inline session/session.go:2244]: why need this function

Comment (glorv) [reply session/session.go]: typo, removed.

**Review by CabinfeverB — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**